### ISSUE TITLE
Force samesite=strict on all cookies

### ISSUE THREAD

**Issue by zenhack** [sandstorm-io/sandstorm#3357]
This was discussed here:

https://github.com/sandstorm-io/sandstorm/pull/2588#issuecomment-581214201

Per the comment, this would block CSRF attacks from domains outside of the sandstorm box's domain (though it might hypothetically still be possible to exploit vulnerabilities in one grain from a different one).

**Comment by zenhack** [comment]
I dug into this a bit, and I'm actually not sure we can do this: it looks like samesite=strict excludes requests originating in an iframe, which means that grains' cookies would not be sent to those grains when viewed through the sandstorm UI.

Found this out by implementing and watching ttrss go into a humorous loop...

**Comment by zenhack** [comment]
For posterity, on the last office hours call we worked out the problem is actually that *.localhost is not considered a subdomain of localhost, so for my local dev setup this was failing, but it does appear to work if I adjust the base url & wildcard hosts appropriately.

In that case we probably need to add some conditional logic to test if the wildcard host is a subdomain of base_url, since if it isn't doing this will break things.

**Comment by zenhack** [comment]
So this will break setups where WILCARD_HOST is not considered a subdomain of BASE_URL -- but chrome sets this by default now, so maybe we should just pull the trigger.

alternatively we could try to avoid breaking such setups by explicitly setting samesite=lax in this case -- but I don't like the idea of silently downgrading people's security if the box is incorrectly configured.

**Comment by ocdtrekkie** [comment]
Could we go ahead and warn people in the UI for sites with setups like the one we saw on IRC with single DNS names that they are misconfigured? Similar to how we warn now about incorrect wildcard setups?

**Comment by zenhack** [comment]
Makes sense to me. Do we want to do something similar (maybe with a
slightly less dire message) for the (probably somewhat more common) case
of WILDCARD_HOST != *.${BASE_URL} ? I think the mitigation is a-priori
worth not supporting that setup.

We should also probably change docs and/or installer to point this issue
out.

Quoting Jacob Weisz (2020-09-17 11:35:05)

**Comment by ocdtrekkie** [comment]
What would be an example of a setup you've seen where the above would be triggered? Also, I assume alpha-*.${BASE_URL} is still a valid methodology, so we need to be more specific: It'd be issues where the wildcard isn't one level sub the base URL.

**Comment by zenhack** [comment]
Right, `alpha-*.${BASE_URL}` is fine. But actually, it looks like alpha is an example of what I'm worried about:

```
BASE_URL=https://alpha.sandstorm.io
WILDCARD_HOST=alpha-*.sandstorm.io
```

/me checks if alpha is broken in latest chromium

The one Wekan grain I have access to seems to still work in Chromium 85. Want to test ttrss on alpha? It was boot-looping when I set SameSite=strict before, so it would be a good test, but I can't create grains on alpha.

**Comment by ocdtrekkie** [comment]
I don't have any Chromium based browsers to test with. I imagine Kenton would be open to letting you access Alpha for testing purposes though.

I understand the example now. And yeah, if we expect them to break we should figure out a way to notify people, and yeah, amend our docs.

**Comment by zenhack** [comment]
//cc @kentonv

**Comment by zenhack** [comment]
TTRSS seems to not be broken on alpha either, so maybe it's okay as long as both the base url and wildcard host are under the same domain?
